tests/models/whisper/test_modeling_whisper.py

```diff
@@ -0,0 +1,851 @@
+            for layer_name, p1 in model_state_dict.items():
+                p2 = loaded_model_state_dict[layer_name]
+                if p1.data.ne(p2.data).sum() > 0:
+                    models_equal = False
+
+            self.assertTrue(models_equal)
+
+
+@require_torch
+@require_torchaudio
+@require_sentencepiece
+@require_tokenizers
+@slow
+class WhisperModelIntegrationTests(unittest.TestCase):
+    @cached_property
+    def default_processor(self):
+        return WhisperProcessor.from_pretrained("openai/whisper-base")
+
+    def _load_datasamples(self, num_samples):
+        from datasets import load_dataset
+
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+        # automatic decoding with librispeech
+        speech_samples = ds.sort("id").select(range(num_samples))[:num_samples]["audio"]
+
+        return [x["array"] for x in speech_samples]
+
+    def test_tiny_logits_librispeech(self):
+
+        from transformers import GPT2Tokenizer, WhisperFeatureExtractor, set_seed
+
+        torch_device = "cpu"
+        set_seed(0)
+        model = WhisperModel.from_pretrained("whisper/tiny")
+        model.to(torch_device)
+
+        # processor = self.default_processor
+
+        input_speech = self._load_datasamples(1)
+
+        feaure_extractor = WhisperFeatureExtractor()
+        tokenizer = GPT2Tokenizer.from_pretrained("gpt2")
+        tokenizer.pad_token = 0
+
+        processor = WhisperProcessor(feaure_extractor, tokenizer)
+
+        input_features = processor(
+            audio=input_speech, text="This part of the speech", return_tensors="pt"
+        ).input_features.to(torch_device)
+        labels = processor(audio=input_speech, text="This part of the speech", return_tensors="pt").labels.to(
+            torch_device
+        )
+        with torch.no_grad():
+            logits = model(
+                input_features,
+                decoder_input_ids=labels,
+                output_hidden_states=False,
+                output_attentions=False,
+                use_cache=False,
+            )
+
+        logits = logits.last_hidden_state @ model.decoder.embed_tokens.weight.T
+
+        # fmt: off
+        EXPECTED_LOGITS = torch.tensor(
+            [
+                8.8958,  4.0423,  9.8841,  9.8493, 10.0628,  4.8472,  9.0100,  5.7364,
+                5.9165,  7.6322,  3.1579, 10.7269,  6.9586, 10.1852,  5.4714,  8.2995,
+                4.7507,  6.6723,  7.2764,  7.1831,  7.0388,  7.2191,  6.2364,  6.2117,
+                5.8797,  2.8099,  6.8319,  5.7094,  0.6999,  6.8444
+            ]
+        )
+        # fmt: on
+        self.assertTrue(torch.allclose(logits[0, 0, :30].cpu(), EXPECTED_LOGITS, atol=1e-4))
+
+    def test_large_logits_librispeech(self):
+
+        from transformers import GPT2Tokenizer, WhisperFeatureExtractor, set_seed
+
+        torch_device = "cpu"
+        set_seed(0)
+        model = WhisperModel.from_pretrained("whisper/large")
```

Comment:
should be `"openai/whisper-large"` when you upload the checkpoins (feel free to already do this :-))